app/presenters/single_content_item_presenter.rb

```diff
@@ -169,24 +164,6 @@ def metadata
     @metadata ||= @single_page_data[:metadata]
   end
 
-  def parse_metrics(time_series_metrics, edition_metrics)
-    metrics = {}
-    time_series_metrics.each do |metric|
-      metrics[metric[:name]] = {
-        'value': metric[:total],
-        'time_series': metric[:time_series]
-      }
-    end
-
-    edition_metrics.each do |metric|
-      metrics[metric[:name]] = {
-        'value': format_metric_value(metric[:name], metric[:value]),
-        'time_series': nil
-      }
-    end
-    metrics
-  end
```

Comment:
Rather than extracting this out in to a class method in a new class, I think it might be possible to push this out of the SingleItemPresenter entirely, and that may end up being simpler.

I think this is the wider context for this class: https://github.com/alphagov/content-data-admin/blob/4a88549f353941a57db1ab1c6f386600d7d9d76b/app/controllers/metrics_controller.rb#L9-L16

It might be neat to try and thread this transformation currently in `parse_metrics` above in to the `FetchSinglePage` class. The already transformed data could then be passed in to the `SingleContentItem` presenter, in the form that it needs. I haven't tried this, but it seems like it could work.

What do you think?

Reply:
@cbaines I really like this idea. Although, I would create a separate PR for it.